File: VB/GridControlWithBar/PropertyGridControlDescendant/SingleRecordViewInfoDescendant.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.XtraVerticalGrid
Imports DevExpress.XtraVerticalGrid.ViewInfo
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.Utils.Drawing
Imports System
Imports System.Collections.Generic



Namespace GridControlWithBar
	Public Class SingleRecordViewInfoDescendant
		Inherits SingleRecordViewInfo
		Public Sub New(ByVal grid As VGridControlBase, ByVal isPrinting As Boolean)
			MyBase.New(grid, isPrinting)

		End Sub



		Protected Overrides Sub CreateBand(ByVal bandHeight As Integer, ByVal bandWidth As Integer)
			Dim left As Integer = (If(ViewRects.BandRects.Count = 0, ViewRects.Client.Left, (CType(ViewRects.BandRects(ViewRects.BandRects.Count - 1), Rectangle)).Right))
			Dim br As New Rectangle(left, ViewRects.Client.Top + (TryCast(Grid, PropertyGridControlDescendant)).BarHeight, bandWidth, Math.Min(bandHeight, ViewRects.Client.Height))
			ViewRects.BandRects.Add(br)

		End Sub

	End Class
End Namespace

File: CS/GridControlWithBar/PropertyGridControlDescendant/SingleRecordViewInfoDescendant.cs
using System.Drawing;
using DevExpress.XtraVerticalGrid;
using DevExpress.XtraVerticalGrid.ViewInfo;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.Utils.Drawing;
using System;
using System.Collections.Generic;



namespace GridControlWithBar
{
    public class SingleRecordViewInfoDescendant : SingleRecordViewInfo
    {
        public SingleRecordViewInfoDescendant(VGridControlBase grid, bool isPrinting)
            : base(grid, isPrinting)
        {

        }


       
        protected override void CreateBand(int bandHeight, int bandWidth)
        {
            int left = (ViewRects.BandRects.Count == 0 ? ViewRects.Client.Left :
          ((Rectangle)ViewRects.BandRects[ViewRects.BandRects.Count - 1]).Right);
            Rectangle br = new Rectangle(left, ViewRects.Client.Top + (Grid as PropertyGridControlDescendant).BarHeight, bandWidth, Math.Min(bandHeight, ViewRects.Client.Height));
            ViewRects.BandRects.Add(br);

        }

    }
}

